### PR TITLE
Flask OAuth

### PR DESCRIPTION
**Pull request by Gouvernathor**
This adds the OAuth feature and everything going with it to the Flask port.

This probably completes the biggest and most comprehensive change done to the repo.

Some quality-of-life changes are made, with feedback in the user interface about the loggedness, about what error were encountered when uploading the diagrams...

It is available on [Beta](https://parliamentdiagram-beta.toolforge.org/).
The only thing required to deploy it on the main website will be a new OAuth consumer with the correct callback.

This should be squash-merged into the flask branch, and then the flask branch squash-merged into master. If no feedback is given, I'll probably wait at least two weeks (so june 10th) if not three (june 17th) before doing that cascade-merge.
In the meantime, "normal" users : please try out the Beta website, and let me know if you see any new bug you're not seeing on the main website, or any change that you don't like. Let me also know if everything works as it should and if you see bugs being fixed !

TODO: (nothing blocking)
- address the question of where to redirect after the callback
  - it probably just needs an optional parameter to the login query, to be saved in the session after a check that it's one of our pages
- ~~maybe remove the toml file from the repo and just list its expected contents in the readme~~

### CODE REVIEW

**Comment by InternauteLambda** [comment]
I tried it just now, and uploading works as normal.

File: src/static/js/arch.js [hunk 1]
```diff
@@ -412,43 +413,207 @@ function updateFilename() {
...
-    //Now get today's date and format it suitably for use in Wikimedia Commons templates:
-    // const today = (new Date()).split("T")[0];
-    let today = new Date();
-    let DD = today.getDate();
-    let MM = today.getMonth() + 1;
-    let YYYY = today.getFullYear();
-
-    if (DD < 10) {
-        DD = '0' + DD
+function makeUploadLink(linkdata, legendtext) {
```
Comment (Gouvernathor) [inline src/static/js/arch.js:416]: Maybe wrap these in `if (oauth_enabled)`

File: src/templates/archinputform.html [hunk 2]
```diff
@@ -4,6 +4,14 @@
 {% block title %}Arch-style parliament diagram generator{% endblock %}
```
Comment (Gouvernathor) [inline src/templates/archinputform.html:4]: the php import at the top of the file needs to be removed

File: src/static/js/arch.js (hunk 1 above)
Comment (Gouvernathor) [reply src/static/js/arch.js:416]: Not necessary

**Pull request merged**